packages/walt-compiler/src/utils/__tests__/__snapshots__/print-node-spec.js.md

```diff
@@ -4,13 +4,111 @@ The actual snapshot is saved in `print-node-spec.js.snap`.
+          (i32.const 1)                                    ;␊
+        )                                                  ;␊
+        (i32.const 2)                                      ;␊
+      )                                                    ;␊
+      (return                                              ;␊
+        (i32.add                                           ;␊
+          (i32.load                                        ;␊
+            (get_local x)                                  ;␊
+            (i32.const 0)                                  ;␊
+          )                                                ;␊
+          (i32.load                                        ;␊
+            (get_local x)                                  ;␊
+            (i32.const 1)                                  ;␊
+          )                                                ;␊
+        )                                                  ;␊
+      )                                                    ;␊
+    )                                                      ;␊
+    `
+
+## imports/exports
+
+> Snapshot 1
+
+    `(import "env" "foo" (type FooType (func (result i32))))    ;␊
+    (import "env" "bar" (type FooType (func (result i32))))    ;␊
+    (type FooType (func (result i32)))                         ;␊
+    (func two (result i32)                                     ;␊
+      (return                                                  ;␊
+        (i32.const 2)                                          ;␊
+      )                                                        ;␊
+    )                                                          ;␊
+    (export                                                    ;␊
+      (func test (result i32)                                  ;␊
+        (return                                                ;␊
+          (call foo                                            ;␊
+            (i32.add                                           ;␊
+              (call two)                                       ;␊
+              (i32.const 2)                                    ;␊
+            )                                                  ;␊
+          )                                                    ;␊
+        )                                                      ;␊
+      )                                                        ;␊
+    )                                                          ;␊
+    `
+
+## plain ast parser
+
+> Snapshot 1
+
+    `(type Type (func param(i32 f32) (result i32)))    ;␊
+    (type-generic Inc)                                ; pseudo type␊
+    (func simple param(y i32) (result i32)            ;␊
+      (local x i32                                    ; immutable␊
+        (i32.const 2)                                 ;␊
+      )                                               ;␊
+      (return                                         ;␊
+        (??.add                                       ;␊
```

Comment:
This should be `i32.add`, right?

Reply:
There are two AST phases in the compiler.  One of them, the first one, doesn't have type information. This is what we see here.

I want equal debug support for both phases though. The printer will omit the types when they aren't yet set.  

Binary expressions are left without types in this phase because walt supports implicit type promotions and eventually type inference. The types of the expression are guaranteed to be balanced.  It's trivial to do this with an additional phase which can apply types based on a full AST.